.github/workflows/e2e-test.yaml

```
@@ -0,0 +1,72 @@
+# This file is part of Astarte.
+#
+# Copyright 2022 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+name: End to End test
+
+on:
+  # Run when pushing to stable branches
+  push:
+    branches:
+      - 'main'
+      - 'release-*'
+  pull_request:
+
+jobs:
+  e2e-test:
+    runs-on: ubuntu-latest
+    env:
+      REALM_NAME: edgehogdevicetest
+      ASTARTE_API_URL: https://api.autotest.astarte-platform.org
+    steps:
+      - name: Create Astarte Cluster
+        id: astarte
+        uses: astarte-platform/astarte-cluster-action@v1
+        with:
+            astarte_realm: $REALM_NAME
+      - name: Checkout sources
+        uses: actions/checkout@v2
+      - name: Checkout edgehog-astarte-interfaces
+        uses: actions/checkout@v2
+        with:
+          repository: edgehog-device-manager/edgehog-astarte-interfaces.git
+          # Update ref when updated interfaces are required.
+          ref: v0.5.2
+          path: ./edgehog/astarte-interfaces
+      - name: Install interface
+        run: | 
+          astartectl realm-management interfaces sync $GITHUB_WORKSPACE/edgehog/astarte-interfaces/*.json --non-interactive
+          astartectl realm-management interfaces ls
+      - name: Register device
+        run: |
+          DEVICE_ID=$(astartectl utils device-id generate-random)
+          echo "E2E_DEVICE_ID=$DEVICE_ID" >> $GITHUB_ENV
+          CREDENTIALS_SECRET=$(astartectl pairing agent register --compact-output "$DEVICE_ID")
```

This will fail when `$DEVICE_ID` starts with `-`. 
```suggestion
          CREDENTIALS_SECRET=$(astartectl pairing agent register --compact-output -- "$DEVICE_ID")
```